### PR TITLE
Add kotlin code for the chapter of tree

### PR DESCRIPTION
**Pull request by curtishd**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [ ] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [ ] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:

- [x] I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
- [x] I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
- [x] The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.
屏幕截图 2024-03-23 193910
屏幕截图 2024-03-23 193926
屏幕截图 2024-03-23 193935
屏幕截图 2024-03-23 193948
屏幕截图 2024-03-23 193956
屏幕截图 2024-03-23 194002
屏幕截图 2024-03-23 194018
屏幕截图 2024-03-23 194027
屏幕截图 2024-03-23 194033
屏幕截图 2024-03-23 194042
屏幕截图 2024-03-23 194101
屏幕截图 2024-03-23 194108

### CODE REVIEW

File: codes/kotlin/chapter_stack_and_queue/array_deque.kt [hunk 1]
```diff
@@ -0,0 +1,144 @@
+/**
+ * File: array_deque.kt
+ * Created Time: 2024-01-25
+ * Author: curtishd (1023632660@qq.com)
+ */
+
+package chapter_stack_and_queue
```
Comment (krahets) [inline codes/kotlin/chapter_stack_and_queue/array_deque.kt:7]: Please remove the code changes of the chapter of stack and queue

**Review by krahets — COMMENTED** [review]
It seems like you didn't push the `utils` code. Could you submit it in another PR?

File: codes/kotlin/chapter_tree/binary_tree.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * Created Time: 2024-01-25
+ * Author: curtishd (1023632660@qq.com)
+ */
+
+package chapter_tree
+
+import utils.TreeNode
+import utils.printTree
+
+fun main() {
```
Comment (krahets) [inline codes/kotlin/chapter_tree/binary_tree.kt:12]: /* Driver Code */

File: codes/kotlin/chapter_tree/array_binary_tree.kt [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+ * Created Time: 2024-01-25
+ * Author: curtishd (1023632660@qq.com)
+ */
+
+package chapter_tree
+
+import utils.TreeNode
+import utils.printTree
+
+class ArrayBinaryTree(private val tree: List<Int?>) {
```
Comment (krahets) [inline codes/kotlin/chapter_tree/array_binary_tree.kt:12]: class header comment

Please check others.

File: codes/kotlin/chapter_stack_and_queue/linkedlist_stack.kt [hunk 4]
```diff
@@ -0,0 +1,86 @@
+/**
+ * File: linkedlist_stack.kt
+ * Created Time: 2024-01-25
+ * Author: curtishd (1023632660@qq.com)
+ */
+
+package chapter_stack_and_queue
+
+class LinkedListStack(
```
Comment (krahets) [inline codes/kotlin/chapter_stack_and_queue/linkedlist_stack.kt:9]: class header comment

(These code had been merged. Please fix them)

**Comment by krahets** [comment]
Thanks for your hard work!

I'll add kotlin code to the docs after merging this PR. And some comments to be addressed:

- `climbing_stiars_dfs_mem.kt` -> `climbing_stairs_dfs_mem.kt`
- Header comments of some classes and functions are missing. Please add them.

**Comment by krahets** [comment]
I've added the kotlin code blocks to the documents at #1170
Please fill in the kotlin code to the documents referring to Python or Java code. Thx!

**Comment by curtishd** [comment]
> It seems like you didn't push the `utils` code. Could you submit it in another PR?

Definitely

**Comment by curtishd** [comment]
> Thanks for your hard work!
> 
> I'll release kotlin code to the web after merging this PR. Before that, some comments should be addressed:
> 
> * `climbing_stiars_dfs_mem.kt` -> `climbing_stairs_dfs_mem.kt`
> * Header comments of some classes and functions are missing. Please add them.
> * Some code files contain an empty line at the bottom but others do not. Did you format all the code?

I will check all the codes again.

File: codes/kotlin/chapter_stack_and_queue/linkedlist_stack.kt (hunk 4 above)
Comment (curtishd) [reply codes/kotlin/chapter_stack_and_queue/linkedlist_stack.kt:9]: ok, I will fix this after this chapter has been merged.

**Pull request closed without merging**